### PR TITLE
FIX API paymentsdistributed accept multicurrency part payment

### PR DESCRIPTION
**Pull request by c3do**
It is now possible to make a partial payment with the multi-currency active.
Fix for #14781 #14783

### CODE REVIEW

File: htdocs/compta/facture/class/api_invoices.class.php [hunk 1]
```diff
@@ -477,16 +477,17 @@ public function postContact($id, $contactid, $type)
 	 *
 	 * @param int    $id             Id of invoice to update
 	 * @param int    $rowid          Row key of the contact in the array contact_ids.
+	 * @param string $type           Type of the contact (BILLING, SHIPPING, CUSTOMER).
 	 *
-	 * @url	DELETE {id}/contact/{rowid}
+	 * @url	DELETE {id}/contact/{rowid}/{type}
```
Comment (eldy) [inline htdocs/compta/facture/class/api_invoices.class.php:482]: Why adding this parameter $type ?
Delete should receive the rowid to delete to be able to delete ?

Comment (c3do) [reply htdocs/compta/facture/class/api_invoices.class.php:482]: If the same contact is on DELIVERY and INVOICE, it is he is unlinked from both positions. It is a real problem if we only have to change the INVOICE contact with a new contact without dissociating the DELIVERY contact. When I unlink a contact on the Dolibarr invoice contact card, I only unlink the billing or the delivery one, it seems logical to me that on the API, this behaves identically.

File: htdocs/compta/facture/class/api_invoices.class.php [hunk 2]
```diff
@@ -1451,41 +1460,60 @@ public function addPaymentDistributed($arrayofamounts, $datepaye, $paymentid, $c
...
-            if ($this->invoice->type == Facture::TYPE_CREDIT_NOTE) {
-                $resteapayer = price2num($resteapayer, 'MT');
-                $amounts[$id] = -$resteapayer;
+
+        	if ($is_multicurrency && $amountarray["multicurrency_amount"] != 'remain')
+        	{
+        	    $amount = price2num($amountarray["multicurrency_amount"], 'MT');
+        	}
+			
+			if ($amount > $remainstopay && $accepthigherpayment == false) {
```
Comment (c3do) [inline htdocs/compta/facture/class/api_invoices.class.php:1500]: fix for #14781

File: htdocs/compta/facture/class/api_invoices.class.php (hunk 1 above)
Comment (c3do) [reply htdocs/compta/facture/class/api_invoices.class.php:482]: @eldy i did not understand your point of view. it is the same reason as orders. https://github.com/Dolibarr/dolibarr/pull/14792#discussion_r493847596

Comment (c3do) [reply htdocs/compta/facture/class/api_invoices.class.php:482]: @eldy it's the same reasoning as with the orders API.

**Pull request merged**